include/deal.II/lac/solver.h

```diff
@@ -21,7 +21,10 @@
 #include <deal.II/lac/vector_memory.h>
 #include <deal.II/lac/solver_control.h>
 
+// Ignore deprecation warnings for auto_ptr.
+DEAL_II_DISABLE_EXTRA_DIAGNOSTICS
 #include <boost/signals2.hpp>
+DEAL_II_ENABLE_EXTRA_DIAGNOSTICS
```

Comment:
Yes, GCC distinguishes between *system* headers and *user* headers, and suppresses some warnings for the former (because users typically don't have control over these header files and can't do anything about it).